src/app/components/hooks/useFetch.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const useFetch = (url: string): any => {
+  const [product, setProduct] = useState(null);
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      const fetchProductData = async () => {
+        const res = await fetch(url);
+        const json = await res.json();
+        setProduct(json);
+      };
+      fetchProductData();
+    }, 200);
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [url]);
```

Comment:
Debounce 🚀

Reply:
It is not working yet :D Still figure out why